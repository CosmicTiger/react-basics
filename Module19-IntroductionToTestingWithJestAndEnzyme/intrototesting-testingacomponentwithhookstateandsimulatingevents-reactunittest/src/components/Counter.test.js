import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import Counter from './Counter';

configure({
    adapter: new Adapter()
});

describe('Testing <Counter /> component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Counter />);
    });

    test('Validate that match with Snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Validate button feature', () => {
        wrapper.find('button').first().simulate('click');

        expect(wrapper.find('h1').text()).toBe('1');

        wrapper.find('button').last().simulate('click');
        wrapper.find('button').last().simulate('click');
        wrapper.find('button').last().simulate('click');

        expect(wrapper.find('h1').text()).toBe('-2');
    });

});