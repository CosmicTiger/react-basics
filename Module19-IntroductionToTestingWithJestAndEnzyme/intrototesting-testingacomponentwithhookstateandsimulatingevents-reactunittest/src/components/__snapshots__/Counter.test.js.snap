// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Counter /> component Validate that match with Snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
  <h1>
    0
  </h1>
</div>
`;
